### ISSUE TITLE
Not appending /1 to some reads

### ISSUE THREAD

**Issue by macmanes** [fstrozzi/Fastool#1]
see https://gist.github.com/macmanes/2ade7dd53835ad633ad4

**Comment by fstrozzi** [comment]
Hi @macmanes , I have took from the Gist the header of the FastQ you mentioned giving the problem, but I am not able to reproduce it. If you can send me a small FastQ that shows this problem, I will investigate. Thanks

**Comment by macmanes** [comment]
OK, try this file: https://www.dropbox.com/s/s6n3umgyldy0ntr/fastools.test.fq

It is extreme bizarre, read HSQ-7001360:67:H88RHADXX:1:1101:17149:2709 is not parsed correctly, though to my eye it looks fine in the input file. Every read after this one, in the output fa file has lost the /1 tag.

**Comment by fstrozzi** [comment]
Hi @macmanes,
I've been able to somewhat reproduce the problem. There is definitely something strange going on in that point of the file and also depending on which gcc and glib version is used to compile Fastool, the issue changes. I've seen disappearing the "/1" as well as the whole nucleotide sequence of the read...I do suspect that _maybe_ there could be some EOF or other strange things in that point of the file (maybe from a cat of multiple files?) that somehow messes up the parser from that point of the file.
I've provided a fix in the latest version, which on my end is now working without issues on your sample dataset. This has been tested against different *nix versions just to be sure:
- Ubuntu 14.04 with GCC version 4.8.2 
- Ubuntu 12.04 with GCC version 4.6.3
- CentOS 6.5 with GCC version 4.4.7
- Mac OSX Maverics 10.9.2

Please try to checkout the latest version and see if this fixes your problem.

**Comment by macmanes** [comment]
OK, done. This indeed works on the test dataset I sent, and I'm testing it on a larger dataset now - assume it will be fine and I'll let you know if something comes up. Thanks for the fix!

I am using Ubuntu 14.04 with gcc 4.8.2. I just upgraded from 13.04 (with gcc 4.7.2, I think) a few days ago!

As you know, these fastq files are always the product of multiple 'cat' events - raw Illumina files are downloaded from the core facility in small 300Mb files, which are combined into a larger file representing a full dataset. So, yes this file has been 'catted', but if unix cat is introducing errors we've got huge problems! Also, the 1st abnormal read was only 1700 reads into the file, with hundreds or thousands of these mis-parsed reads in total. So, I'd be surprised if this were a cat related issue.
